adapters/handlers/rest/authz/handlers_authz.go

```diff
@@ -348,7 +348,7 @@ func (h *authZHandlers) assignRoleToUser(params authz.AssignRoleToUserParams, pr
 		return authz.NewAssignRoleToUserNotFound()
 	}
 
-	if err := h.controller.AddRolesForUser(params.ID, params.Body.Roles); err != nil {
+	if err := h.controller.AddRolesForUser(conv.PrefixUserName(params.ID), params.Body.Roles); err != nil {
```

Comment:
is this backward compatible?

Reply:
what you mean, it's internal, shall not affect anything

Reply:
if users added roles to users before and upgrade. Will it still work?

Reply:
yes, `PrefixUserName()` is no-op if the id was already prefixed 
see https://github.com/weaviate/weaviate/blob/b86330e680ccc584057d1ab6a40282632cf88f45/usecases/auth/authorization/conv/casbin_types.go#L400

Reply:
alright 👍